### PR TITLE
[usage] Batch list workspace instances in range

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Lists Workspace Instances in batches of 100

## Related Issue(s)

Fixes #

## How to test

Unit tests

## Release Notes

```release-note
NONE
```

## Documentation

NONE

### CODE REVIEW

File: components/usage/pkg/db/workspace_instance.go [hunk 1]
```diff
@@ -70,13 +70,17 @@ func (i *WorkspaceInstance) TableName() string {
...
 	var instances []WorkspaceInstance
+	var instancesInBatch []WorkspaceInstance
 	tx := conn.WithContext(ctx).
 		Where(
 			conn.Where("stoppedTime >= ?", TimeToISO8601(from)).Or("stoppedTime = ?", ""),
 		).
 		Where("creationTime < ?", TimeToISO8601(to)).
 		Where("startedTime != ?", "").
-		Find(&instances)
+		FindInBatches(&instancesInBatch, 1000, func(tx *gorm.DB, batch int) error {
```
Comment (andrew-farries) [inline components/usage/pkg/db/workspace_instance.go:80]: What's the `batch` parameter intended to be used for? If it's not used can we replace it with `_` ?

File: components/usage/pkg/db/workspace_instance_test.go [hunk 2]
```diff
@@ -191,3 +191,29 @@ func TestListWorkspaceInstancesInRange(t *testing.T) {
...
+			ID:           uuid.New(),
+			WorkspaceID:  workspaceID,
+			CreationTime: db.NewVarcharTime(time.Date(2022, 05, 15, 12, 00, 00, 00, time.UTC)),
+			StartedTime:  db.NewVarcharTime(time.Date(2022, 05, 15, 12, 00, 00, 00, time.UTC)),
+			StoppedTime:  db.NewVarcharTime(time.Date(2022, 05, 15, 13, 00, 00, 00, time.UTC)),
+		}))
+
+	}
+
+	tx := conn.CreateInBatches(&instances, 1000)
```
Comment (andrew-farries) [inline components/usage/pkg/db/workspace_instance_test.go:211]: Why do we need to `CreateInBatches` here?

Comment (easyCZ) [reply components/usage/pkg/db/workspace_instance_test.go:211]: We don't strictly, but felt like I should start a pattern of not doing things with more than 1k entries outside batches

File: components/usage/pkg/db/workspace_instance.go (hunk 1 above)
Comment (easyCZ) [reply components/usage/pkg/db/workspace_instance.go:80]: https://gorm.io/docs/advanced_query.html#FindInBatches

Will make it unused.

File: components/usage/pkg/db/dbtest/workspace_instance.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ func NewWorkspaceInstance(t *testing.T, instance db.WorkspaceInstance) db.Worksp
 		id = instance.ID
 	}
 
-	workspaceID := generateWorkspaceID()
+	workspaceID := GenerateWorkspaceID()
```
Comment (easyCZ) [inline components/usage/pkg/db/dbtest/workspace_instance.go:27]: This also fixes broken master due to a semantic merge.

**Review by andrew-farries — APPROVED** [review]

**Pull request merged**